### DISCUSSION TITLE
Lazygit File History View: How to get the History|Preview split view by default?

### DISCUSSION THREAD

**Discussion by alewis001** [LazyVim/LazyVim#4650] [Q&A] [answered]
In LazyGIt (outside of Lazyvim), if I go to the Commits panel, do ctrl+s and find a file, the view that appears is "Commits / Reflog" on the left and  "Patch" on the right (showing the commit patch preview for the file I selected in the filter). 

In LazyVIm, using `<leader>gf` I get just the "Commits / Reflog" panel. I can use the `+` button to cycle through the views to get to the one above but I was wondering whether if there is a way to make that split view the default view for `<leader>gf`?

I can't see a command-line option for invoking lazygit along with the `-f` flag, which is what I think Lazyvim does to bring up the filtered view. There also doesn't appear to be a config setting for the filter view that controls the half/full view mode, as far as I can tell.

I wanted to check in case I'm missing something before I raise an issue in LazyGit (and if I get time try to contribute a change). I think there's also a way to implement custom commands in LG so maybe that's an option as well.

Cheers

**Comment by dpetka2001** [comment]
The `-f` flag gets passed directly into the [mapping](https://github.com/LazyVim/LazyVim/blob/cb40a09538dc0c417a7ffbbacdbdec90be4a792c/lua/lazyvim/config/keymaps.lua#L138-L141) as you can see.

**Reply by alewis001** [reply]
Sorry, yeah, that's what I meant by...

> I can't see a command-line option for invoking lazygit along with the -f flag

That there isn't another flag that can be applied along with the `-f` flag in order to get the behaviour I'm looking for. My hope was that lazygit would have had a command-line arg for specifying the view mode (I think they might be 'normal' | 'half' | 'full') so I would have overridden `<leader>gf` with effectively a copy of what's in the mapping you linked to but with the extra argument to specify the view mode. Unfortunately, no such "view mode" argument exits in LG as far as I can tell.

**Reply by dpetka2001** [reply, marked as answer]
Better ask in lazygit repo if such a feature is available with some kind of flag. They will know better.

**Reply by alewis001** [reply]
Yeah, thanks. I just wanted to make sure I wasn't missing something and someone says "ah yeah, you can do that, just do..." :) 

I'll ask around in Lazygit and mark this as resolved.

Thanks again.

**Comment by alewis001** [comment]
In case it's helpful to someone else, I've started a discussion in lazygit here: https://github.com/jesseduffield/lazygit/discussions/4011

**Reply by alewis001** [reply]
My PR for a change to add a `-sm normal|half|full` (or `--screen-mode normal|half|full`) command line argument has been merged into Lazygit: https://github.com/jesseduffield/lazygit/pull/4103.

So if you want LG to load the file history view in `half` mode like I do, then I think you should be able to modify your lazyvim config to add the command line arg once LG has a release with my changes included.
